File: week-13/bkryusim/체육복.py
def solution(n, lost, reserve):
    answer = n - len(lost)

    for l in lost:
        for r in reserve:
            if l == r:
                answer += 1
                lost.pop(lost.index(l))
                reserve.pop(reserve.index((r)))
                break

    lost.sort()
    reserve.sort()

    for l in lost:
        for i, r in enumerate(reserve):
            if r == l + 1 or r == l - 1 or r == l:
                reserve[i] = -1
                answer += 1
                break

    return answer

File: week-13/protoseo/체육복.java
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int[] count = new int[n + 2];
        for (int i : lost) {
            count[i]--;
        }
        for (int i : reserve) {
            count[i]++;
        }
        for (int i = 1; i <= n; i++) {
            if (count[i] == -1) {
                if (count[i - 1] > 0) {
                    count[i]++;
                    count[i - 1]--;
                } else if (count[i + 1] > 0) {
                    count[i]++;
                    count[i + 1]--;
                }
            }
        }
        
        int answer = 0;
        for (int i = 1; i <= n; i++) {
            if (count[i] > -1) {
                answer++;
            }
        }
        return answer;
    }
}